staff/bernat-casasus/restaurants-app.1/scripts/main.js

```diff
@@ -0,0 +1,145 @@
+function SearchPanel() {
+    Component.call(this, 'form');
+
+    var input = document.createElement('input');
+    input.type = 'search';
+    input.placeholder = 'Input a text...';
+
+    var button = document.createElement('button');
+    button.type = 'submit';
+    button.innerHTML = 'Search';
+
+    this.element.appendChild(input);
+    this.element.appendChild(button);
+
+    var _callback;
+
+    this.element.addEventListener('submit', function (event) {
+        event.preventDefault();
+
+        var query = input.value;
+
+        if (query && _callback) _callback(query);
+    }.bind(this));
+
+    this.onSearch = function (callback) {
+        _callback = callback;
+    };
+}
+
+SearchPanel.prototype = Object.create(Component.prototype);
+SearchPanel.prototype.constructor = SearchPanel;
+
+var search = new SearchPanel();
+
+
+search.onSearch(function (query) {
+    
+    logic.searchBeers(query, function(beer){ 
+
+
+        results.updateResults(beer.map(function (result) {
```

Comment:
beer in restaurant again

Reply:
Please If posible use english to send comments! Yeah it is the testing version.